data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java

```diff
@@ -250,7 +249,8 @@ private boolean egressEquals(DataPlaneContract.Egress e1, DataPlaneContract.Egre
       && Objects.equals(e1.getReplyToOriginalTopic(), e2.getReplyToOriginalTopic())
       && Objects.equals(e1.getEgressConfig(), e2.getEgressConfig())
       && Objects.equals(e1.getFilter(), e2.getFilter())
-      && Objects.equals(e1.getKeyType(), e2.getKeyType());
+      && Objects.equals(e1.getKeyType(), e2.getKeyType())
+      && Objects.equals(e1.getDialectedFilterList(), e2.getDialectedFilterList());
```

Comment:
Can we add a test for this change detection?

Reply:
Done